elide-core/src/main/java/com/yahoo/elide/core/EntityPermissions.java

```diff
@@ -75,12 +75,17 @@ public EntityPermissions(EntityDictionary dictionary,
                              Collection<AccessibleObject> fieldOrMethodList)  {
         this.dictionary = dictionary;
         for (Class<? extends Annotation> annotationClass : PERMISSION_ANNOTATIONS) {
-            ParseTree classPermission = bindClassPermissions(cls, annotationClass);
             final Map<String, ParseTree> fieldPermissions = new HashMap<>();
             fieldOrMethodList.stream()
                     .forEach(member -> bindMemberPermissions(fieldPermissions, member, annotationClass));
-            if (classPermission != null || !fieldPermissions.isEmpty()) {
-                bindings.put(annotationClass, new AnnotationBinding(classPermission, fieldPermissions));
+            if (annotationClass == SharePermission.class
+                    && EntityDictionary.getFirstAnnotation(cls, Arrays.asList(annotationClass)) != null) {
+                bindings.put(SharePermission.class, new AnnotationBinding(null, fieldPermissions));
```

Comment:
Should we be binding the read permissions here?

Reply:
If I understand it right, I think we still bind SharePermission here to indicate if the entity is shareable or not, and when SharePermission evaluation is trigger in `PersistenceResource` we then fall back to evaluate the ReadPermission of entity.